app/models/entity.rb

```diff
@@ -577,6 +577,15 @@ def unique_addresses
     index.values.concat(nils)
   end
 
+  def total_usd_donations
```

Comment:
You don't need to join with RelationshipCategory:
``` rb
.where(category_id: Relationship::DONATION_CATEGORY)
.where(currency: 'USD')
.sum(:amount)
```

Reply:
It seems a bit wrong to me to rely on a constant that has the relevant ID hard-coded. Shouldn't proper separation of concerns between database and logic prohibit this?